adb-client.js

```diff
@@ -14,7 +14,13 @@ const { AdbSocket } = require("./adb-socket");
 
 Cu.import("resource://gre/modules/Services.jsm");
 
-let { TextEncoder, TextDecoder } = Cu.import("resource://gre/modules/Services.jsm");
+let services =
+  Cu.import("resource://gre/modules/Services.jsm", {});
+// Starting with FF57, jsm share the same global and this require pulling it from it.
+if (!services.TextEncoder) {
+  services = Cu.getGlobalForObject(services);
+}
+let { TextEncoder, TextDecoder } = services;
```

Comment:
The alternative according to comment https://bugzilla.mozilla.org/show_bug.cgi?id=1401373#c7 is

```javascript
let { TextDecoder, TextDecoder } = Cu.getGlobalForObject(Cu.import("resource://gre/modules/Services.jsm", {}));
```

Knowing that this also works pre57, how do you feel about only using this (keeping you comment of course).

I noticed we are also importing Services.jsm as a global 2 lines above. We should have access to TextEncoder and TextDecoder on the current global already I think?